Doc/tools/version_next.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+"""
+Replace `.. versionchanged:: next` lines in docs files by the given version.
+
+Applies to all the VersionChange directives. For deprecated-removed, only
+handle the first argument (deprecation version, not the removal version).
+
+"""
+
+import re
+import sys
+import argparse
+from pathlib import Path
+
+DIRECTIVE_RE = re.compile(
+    r'''
+        (?P<before>
+            \s*\.\.\s+
+            (version(added|changed|removed)|deprecated(-removed)?)
+            \s*::\s*
```

Comment:
Whilst it is 'allowed', there really shouldn't be space before the '::'.
```suggestion
            (version(added|changed|removed)|deprecated(-removed)?)::\s*
```

Reply:
IMO, that's a job for a linter. In the script I want to catch everything Sphinx might consider a version.